read.go

```diff
@@ -234,6 +234,12 @@ func (r *Reader) ReadLiteral() (Literal, error) {
 		return nil, err
 	}
 	lstr = trimSuffix(lstr, literalEnd)
+
+	nonSync := lstr[len(lstr)-1] == '+'
```

Comment:
This doesn't check whether the string is non-empty. We could use `strings.HasSuffix` to prevent issues like this.

Reply:
Oops.